packages/functional-tests/tests/syncV3/syncV3ResetPassword.spec.ts

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { test, expect } from '../../lib/fixtures/standard';
+import { EmailHeader, EmailType } from '../../lib/email';
+
+const PASSWORD = 'passwordzxcv';
+
+test.describe('Firefox Desktop Sync v3 reset password', () => {
+  test.beforeEach(({}, testInfo) => {
+    test.slow(testInfo.project.name !== 'local', 'email delivery can be slow');
+  });
+
+  test('reset password, verify same browser', async ({
+    target,
+    page,
+    pages: { login, resetPassword },
+  }) => {
+    const uaStrings = {
+      desktop_firefox_58:
+        'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:58.0) Gecko/20100101 Firefox/58.0',
+    };
+    const query = { forceUA: uaStrings['desktop_firefox_58'] };
+    const queryParam = new URLSearchParams(query);
+    const email = login.createEmail();
+    await target.auth.signUp(email, PASSWORD, {
+      lang: 'en',
+      preVerified: 'true',
+    });
+    await login.clearCache();
+    await page.goto(
+      `${
+        target.contentServerUrl
+      }/reset_password?context=fx_desktop_v3&service=sync/?${queryParam.toString()}`
```

Comment:
Relevant to nothing, but I was just poking this and seeing how the dual `?` in the query string would parse. 
Here's a sample of a mock https://localhost URL (both as a string and URL object). So it will pass two query params, `context` and  `service`; where `service` will look like `'sync/?forceUA=Mozilla/5.0 …'`:

```sh
https://localhost/reset_password?context=fx_desktop_v3&service=sync/?forceUA=Mozilla%2F5.0+%28Windows+NT+6.1%3B+WOW64%3B+rv%3A58.0%29+Gecko%2F20100101+Firefox%2F58.0

URL {
  href: 'https://localhost/reset_password?context=fx_desktop_v3&service=sync/?forceUA=Mozilla%2F5.0+%28Windows+NT+6.1%3B+WOW64%3B+rv%3A58.0%29+Gecko%2F20100101+Firefox%2F58.0',
  origin: 'https://localhost',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'localhost',
  hostname: 'localhost',
  port: '',
  pathname: '/reset_password',
  search: '?context=fx_desktop_v3&service=sync/?forceUA=Mozilla%2F5.0+%28Windows+NT+6.1%3B+WOW64%3B+rv%3A58.0%29+Gecko%2F20100101+Firefox%2F58.0',
  searchParams: URLSearchParams {
    'context' => 'fx_desktop_v3',
    'service' => 'sync/?forceUA=Mozilla/5.0 (Windows NT 6.1; WOW64; rv:58.0) Gecko/20100101 Firefox/58.0' },
  hash: ''
}
```